### PR TITLE
torizonPackages:Separate buildDeps from devDeps(renamed to devRuntime…

### LINKED ISSUE #205: C++ CMake Console Application Template has no torizonPackages.json autocomplete feature
The [Ctrl+Space Autocomplete](https://developer.toradex.com/torizon/application-development/ide-extension/add-dependencies-and-tools-to-existing-project/#add-debian-packages-with-torizonpackagesjson) for packages on the `torizonPackages.json` file is not working for the ([C++ CMake Console Application](https://github.com/toradex/vscode-torizon-templates/tree/dev/cmakeConsole)) template.

When Ctrl+Space is pressed, it only gives suggestions for the local words in the file:

Screenshot_20240524_100047

The template also has the following warning for the `torizonPackages.json` file, which seems to be related to this problem:

```
Unable to load schema from 'debian://schemas/packages': cannot open debian://schemas/packages. Detail: Unable to resolve resource debian://schemas/packages.
```

### PR DESCRIPTION
**Pull request by andreriesco**
…Deps)

torizonPackages.json: Separate the dependencies used to build/compile the code (the ones on Dockerfile.sdk and on the build part of the Dockerfile), which are now on buildDeps, from the runtime dependencies, which are now on devRuntimeDeps(changed from devDeps) for the Dockerfile.debug and on prodRuntimeDeps (changed from deps) for the runtime part of the Dockerfile.

Fixes: #205

### CODE REVIEW

File: scripts/torizonPackages.ps1 [hunk 1]
```diff
@@ -113,17 +143,18 @@ if (Test-Path -Path "Dockerfile.sdk") {
...
 
 # Dockerfile
 Write-Host "Applying to Dockerfile ..."
 $Dockerfile = _getFileLines "Dockerfile"
-$Dockerfile = _ReplaceSection $Dockerfile "torizon_packages_prod"
-_ReplaceSection $Dockerfile "torizon_packages_dev" `
+$Dockerfile = `
+_ReplaceSection $Dockerfile "torizon_packages_prod"
+$Dockerfile = `
+_ReplaceSection $Dockerfile "torizon_packages_build" `
```
Comment (microhobby) [inline scripts/torizonPackages.ps1]: Please indent these

File: assets/tasks/common.json [hunk 2]
```diff
@@ -478,10 +478,12 @@
                 "showReuseMessage": true,
                 "clear": false
             },
+            "env": {
+                "TORIZON_ARCH": "${config:torizon_arch}"
+            },
             "args": [
                 "-nop",
-                "${workspaceFolder}/.conf/torizonPackages.ps1",
-                "${config:torizon_arch}"
```
Comment (microhobby) [inline assets/tasks/common.json]: so, the `TORIZON_ARCH` here is the first argument that is being sent by the `${config:torizon_arch}`. So, I do not get why this change is needed.

File: assets/json/torizonPackages.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
-    "deps": [
+    "prodRuntimeDeps": [
     ],
-    "devDeps": [
+    "devRuntimeDeps": [
```
Comment (microhobby) [inline assets/json/torizonPackages.json:4]: what `Runtime` means?

File: .github/workflows/build-python.yaml [hunk 4]
```diff
@@ -31,6 +31,10 @@ jobs:
 
         run: |
           docker run --rm --privileged torizon/binfmt
+          if ($env:PROJECT_ARCH -eq "arm64") { $env:TORIZON_ARCH = "aarch64" }
+          if ($env:PROJECT_ARCH -eq "armhf") { $env:TORIZON_ARCH = "armhf" }
+          echo "TORIZON_ARCH=$env:TORIZON_ARCH" | `
+            Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
```
Comment (microhobby) [inline .github/workflows/build-python.yaml]: is not easier only set the `config:torizon_arch`?

File: javaForms/Dockerfile [hunk 5]
```diff
@@ -41,8 +41,8 @@ RUN apt-get -q -y update && \
     msopenjdk-21 \
     # ADD YOUR PACKAGES HERE
 # DO NOT REMOVE THIS LABEL: this is used for VS Code automation
-    # __torizon_packages_dev_start__
-    # __torizon_packages_dev_end__
+    # __torizon_packages_build_start__
+    # __torizon_packages_build_end__
```
Comment (microhobby) [inline javaForms/Dockerfile:45]: the string here mention `build` but on the `torizonPackages.json` it's mention "devRuntimeDeps", it does not maintain consistency.

**Review by microhobby — CHANGES_REQUESTED** [review]

File: assets/json/torizonPackages.json (hunk 3 above)
Comment (andreriesco) [reply assets/json/torizonPackages.json:4]: The one used during the execution of the application

File: javaForms/Dockerfile (hunk 5 above)
Comment (andreriesco) [reply javaForms/Dockerfile:45]: Not sure I understood this one

File: .github/workflows/build-python.yaml (hunk 4 above)
Comment (andreriesco) [reply .github/workflows/build-python.yaml]: I think so, I will try to use it

File: assets/tasks/common.json (hunk 2 above)
Comment (andreriesco) [reply assets/tasks/common.json]: I changed because CI/CD was failing, but I will try to pass `config:torizon_arch` directly on CI/CD

File: javaForms/Dockerfile (hunk 5 above)
Comment (microhobby) [reply javaForms/Dockerfile:45]: like, the `__torizon_packages_dev_start__` is not build? Something is not "glueing" the name on the JSON and here on the headers.

Comment (andreriesco) [reply javaForms/Dockerfile:45]: Before this commit it was `__torizon_packages_dev_start__` for the debug runtime container and also for the build/compilation containers (Dockerfile.sdk and build part of Dockerfile). Now  `__torizon_packages_dev_start__` is just for the debug runtime container (`devRuntimeDeps`) and `__torizon_packages_build_start__` for the build/compilation containers (`buildDeps`)

**Comment by microhobby** [comment]
LGTM
